api/server/router/build/build.go

```diff
@@ -5,14 +5,13 @@ import "github.com/docker/docker/api/server/router"
 // buildRouter is a router to talk with the build controller
 type buildRouter struct {
 	backend Backend
+	daemon  daemonBackend
 	routes  []router.Route
 }
 
 // NewRouter initializes a new build router
-func NewRouter(b Backend) router.Router {
-	r := &buildRouter{
-		backend: b,
-	}
+func NewRouter(b Backend, d daemonBackend) router.Router {
```

Comment:
I think `daemonBackend` is more code smell as nothing should depend on "daemon". "daemon" just means anything that runs in the background. The router should depend on features. So if you want to keep the interface this should be `experimentalProvider`. If you just don't like booleans for some reason you can make an opt struct.